src/dafny/Collections/Arrays.dfy

```diff
@@ -15,7 +15,7 @@ module Dafny.Collections.Arrays {
   import opened Relations
   import opened Seq
 
-  method BinarySearch<T>(a: array<T>, key: T, less: (T, T) -> bool) returns (r: Option<nat>)
+  method BinarySearch<T(!new)>(a: array<T>, key: T, less: (T, T) -> bool) returns (r: Option<nat>)
```

Comment:
I'm very surprised by this type characteristic. I would have thought it would have been possible to sort an array of arrays by the size of each array for example. Why is this required now? Can we relax it if possible?

Reply:
It's because of how the specification was written. The precondition uses `StrictTotalOrdering`, which ends up quantifying over all values of the type. If those ordering predicates were instead written to apply only to an element of given set or sequence, then the `(!new)` would not be needed.

Reply:
Bumping on this one.

Reply:
When I wrote my last message, I hadn't seen yours on the thread. I should have refreshed the page first.